### PR TITLE
Experimental Webpack support

### PR DESCRIPTION
**Pull request by glebm**
Refs #831

@ahorek @dazzaroonie Please take a look

thredded_create_app PR: https://github.com/thredded/thredded_create_app/pull/17

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,6 +4,7 @@ source 'https://rubygems.org'
 
 gem 'rails', '~> 6.0.0'
 gem 'rails-i18n', '~> 6.0.0'
+gem 'webpacker', '~> 4.0'
```
Comment (ahorek) [inline Gemfile]: the gem should be also in https://github.com/thredded/thredded/blob/master/spec/gemfiles/rails_6_0.gemfile ?

ideally you should test both variants
1/ rails 6 + webpack
2/ rails 6 + sprockets 4

File: spec/dummy/config/initializers/assets.rb [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # frozen_string_literal: true
 
-if Gem::Version.new(Sprockets::VERSION) < Gem::Version.new('4')
-  Rails.application.config.assets.precompile += %w[manifest.js]
+Rails.application.config.assets.precompile +=
+  if Gem::Version.new(Sprockets::VERSION) < Gem::Version.new('4')
+    %w[manifest.js]
+  else
+    %w[day.css night.css email.css]
```
Comment (ahorek) [inline spec/dummy/config/initializers/assets.rb]: this shouln't be necessary, the manifest file is already included https://github.com/rails/sprockets-rails/blob/master/lib/sprockets/railtie.rb#L107

File: spec/dummy/config/application.rb [hunk 3]
```diff
@@ -29,6 +29,11 @@
 
 require 'web-console' if Rails.env.development?
 
+# If true, use Webpack. Otherwise, use Sprockets.
+THREDDED_TESTAPP_WEBPACK = Rails::VERSION::MAJOR >= 6 && ENV['THREDDED_TESTAPP_SPROCKETS'] != '1'
```
Comment (ahorek) [inline spec/dummy/config/application.rb]: you should avoid using constants in a global scope, I would prefer a helper method

Comment (glebm) [reply spec/dummy/config/application.rb]: Done. Some duplication but whatever.

File: spec/dummy/config/initializers/assets.rb (hunk 2 above)
Comment (glebm) [reply spec/dummy/config/initializers/assets.rb]: Done

File: spec/dummy/config/initializers/assets.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # frozen_string_literal: true
 
 if Gem::Version.new(Sprockets::VERSION) < Gem::Version.new('4')
-  Rails.application.config.assets.precompile += %w[manifest.js]
+  Rails.application.config.assets.precompile += %w[day.css night.css email.css]
```
Comment (ahorek) [inline spec/dummy/config/initializers/assets.rb]: just precompile manifest.js for sprockets < 4 (sprockets 4 includes it by default), this way you don't have to define top level assets at two places

File: Gemfile (hunk 1 above)
Comment (glebm) [reply Gemfile]: Done

**Pull request merged**